### PR TITLE
(IR interpreter) Add more intrinsics

### PR DESCRIPTION
**Pull request by protonu**
Add support for more intrinsics in the IR eval path. Add more valid ops to guard_elimination.cpp

### CODE REVIEW

**Review by bertmaher — COMMENTED** [review]
Minor comment: it'd be a lot easier to review this PR if you didn't roll the formatter changes into the PR.  I know GitHub kind of sucks at this because there's no real workflow for stacked PRs, but yeah.

**Review by bertmaher — COMMENTED** [review]
Also can you rebase this, I think the "fixes to build on system..." commit is already in trunk

File: torch/csrc/jit/passes/tensorexpr_fuser.cpp [hunk 1]
```diff
@@ -736,34 +760,33 @@ class TensorExprKernel {
...
-            new_axes[dim] = new_axes[dim] - IntImm::make(offset);
-
-            for (int ii = 1; ii < inputs.size(); ++ii) {
-              load = ifThenElse(
-                CompareSelect::make(axes[dim], IntImm::make(offset), kLT),
-                load,
-                tensorOrConstant(inputs[ii], new_axes)
-              );
-              offset += bufferSizes(tensors_.at(inputs[ii]->unique()))[dim];
+            "aten_cat", texprDims(v), [this, v](const std::vector<Var>& axes) {
```
Comment (bertmaher) [inline torch/csrc/jit/passes/tensorexpr_fuser.cpp:763]: Are these just formatting changes?  Please revert formatting changes that aren't part of your functional diff.

File: torch/csrc/jit/passes/tensorexpr_fuser.cpp [hunk 2]
```diff
@@ -805,7 +828,8 @@ class TensorExprKernel {
 #endif
 #ifdef ENABLE_LLVM
       case kLLVMCodeGen:
-        codegen_ = std::make_unique<LLVMCodeGen>(stmt, params);
+        codegen_ =
+            std::make_unique<torch::jit::tensorexpr::LLVMCodeGen>(stmt, params);
```
Comment (bertmaher) [inline torch/csrc/jit/passes/tensorexpr_fuser.cpp:832]: Why is the fully qualified name necessary here?

**Comment by protonu** [comment]
Let me close this and start a clean PR.

**Pull request closed without merging**